COM-597 Program module - add a new search list for HIV defaulter tracking

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-597

tmvumbi2 committed:
COM-597 Send language as a query parameter to back-end calls
This parameter will be used in the SQL queries to retrieve the correct locale for concept names

lauravignoli commented on ui/app/common/patient-search/controllers/patientsListController.js:
```diff
@@ -40,7 +40,8 @@ angular.module('bahmni.common.patientSearch')
                 var params = { q: searchType.handler, v: "full",
                     location_uuid: $bahmniCookieStore.get(Bahmni.Common.Constants.locationCookieName).uuid,
-                    provider_uuid: $rootScope.currentProvider.uuid };
+                    provider_uuid: $rootScope.currentProvider.uuid,
```
@tmvumbi2 The tests are failing, please fix.

lauravignoli replied:
TypeError: $rootScope.currentUser is undefined in /home/travis/build/jembi/cameroon-openmrs-module-bahmniapps/ui/app/common/patient-search/controllers/patientsListController.js (line 9)

tmvumbi2 committed:
COM-597 Fix breaking tests
Set the language value only when the variable currentUser.userProperties is defined in $rootScope.

tmvumbi2 committed:
COM-597 Fix lint bug
